styx-common/src/main/java/com/spotify/styx/model/Workflow.java

```diff
@@ -37,7 +38,7 @@
   public abstract String workflowId();
 
   @JsonProperty
-  public abstract URI componentUri();
+  public abstract Optional<URI> componentUri();
```

Comment:
Maybe add a slight bit of documentation to this? Even if of the "legacy, don't use" sort.